### PR TITLE
i3s-debug: added selector for changing bounding volumes geometry

### PR DESCRIPTION
**Pull request by morphoix**

### CODE REVIEW

File: examples/experimental/i3s-17-and-debug/app-debug.js [hunk 1]
```diff
@@ -31,7 +31,8 @@ import {
   INITIAL_MAP_STYLE,
   CONTRAST_MAP_STYLES,
   INITIAL_TILE_COLOR_MODE,
-  INITIAL_BOUNDING_VOLUME_COLOR_MODE
+  INITIAL_BOUNDING_VOLUME_COLOR_MODE,
+  INITIAL_BOUNDING_VOLUME_TYPE_MODE
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/app-debug.js:35]: `INITIAL_BOUNDING_VOLUME_TYPE`

File: examples/experimental/i3s-17-and-debug/bounding-volume-layer.js [hunk 2]
```diff
@@ -91,19 +89,33 @@ export default class BoundingVolumeLayer extends CompositeLayer {
     return geometry;
   }
 
-  _generateMesh(tile) {
-    if (tile.header.obb || tile.boundingVolume instanceof OrientedBoundingBox) {
-      return this._generateCubeMesh(tile);
+  _generateMesh(tile, boundingVolumeType) {
+    switch (boundingVolumeType) {
+      case 'CubeGeometry':
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/bounding-volume-layer.js:94]: Perhaps always generate both and just choose the one we need when rendering?

File: examples/experimental/i3s-17-and-debug/constants.js [hunk 3]
```diff
@@ -37,3 +37,10 @@ export const BOUNDING_VOLUME_WARNING_TYPE = 'boundingVolume';
 export const LOD_WARNING_TYPE = 'lod';
 export const PARENT_LOD_WARNING_TYPE = 'parentLod';
 export const GEMETRY_VS_TEXTURE_WARNING_TYPE = 'geometryVsTextures';
+
+export const BOUNDING_VOLUME_SHAPE_MODE = {
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/constants.js:41]: When are these constants used? If presented to the user, should we use the terminology `Bounding Sphere` and `Oriented Bounding Box`?

**Review by ibgreen — APPROVED** [review]

File: examples/experimental/i3s-17-and-debug/constants.js (hunk 3 above)
Comment (ibgreen) [reply examples/experimental/i3s-17-and-debug/constants.js:41]: `BOUNDING_VOLUME_TYPE` ?

File: examples/experimental/i3s-17-and-debug/bounding-volume-layer.js (hunk 2 above)
Comment (morphoix) [reply examples/experimental/i3s-17-and-debug/bounding-volume-layer.js:94]: Ok, I'll create another PR to fix it

File: examples/experimental/i3s-17-and-debug/constants.js (hunk 3 above)
Comment (morphoix) [reply examples/experimental/i3s-17-and-debug/constants.js]: it's now fixed with names of constants

File: examples/experimental/i3s-17-and-debug/components/debug-panel.js [hunk 4]
```diff
@@ -351,8 +395,7 @@ export default class DebugPanel extends PureComponent {
         <DropDown
           value={tileColorMode}
           onChange={evt => {
-            const selected = evt.target.value;
-            this._onChangedTileColorMode({tileColorMode: parseInt(selected, 10)});
+            this._onChangedTileColorMode({tileColorMode: parseInt(evt.target.value, 10)});
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/components/debug-panel.js:398]: Ditto

File: examples/experimental/i3s-17-and-debug/components/debug-panel.js [hunk 5]
```diff
@@ -273,13 +293,13 @@ export default class DebugPanel extends PureComponent {
...
           <Label htmlFor="boundingVolume">Show</Label>
+          {boundingVolume ? this._renderBoundingTypes() : null}
         </CheckboxOption>
         <DropDown
           value={boundingVolumeColorMode}
           onChange={evt => {
-            const selected = evt.target.value;
             this._onChangedBoundingVolumeColorMode({
-              boundingVolumeColorMode: parseInt(selected, 10)
+              boundingVolumeColorMode: parseInt(evt.target.value, 10)
```
[the remark this answers is not in the document]
Comment (morphoix) [reply examples/experimental/i3s-17-and-debug/components/debug-panel.js:302]: > const boundingVolumeColorMode = parseInt(evt.target.value, 10);
> this._onChangedBoundingVolumeColorMode({boundingVolumeColorMode});        
> ```
we have constants with such a names in the state assignment so we can not rename it

**Pull request merged**